packages/web/components/complex/paginated-table.tsx

```diff
@@ -0,0 +1,134 @@
+import { flexRender, Row, Table } from "@tanstack/react-table";
+import { useVirtualizer } from "@tanstack/react-virtual";
+import Image from "next/image";
+import Link from "next/link";
+import { useRef } from "react";
+import { IS_FRONTIER } from "../../config";
+import { Pool } from "./all-pools-table-set";
+
+const SIZE = 80;
+
+type Props = {
+  paginate: () => void;
+  table: Table<Pool>;
+};
+
+const PaginatedTable = ({ table }: Props) => {
+  const tableContainerRef = useRef<HTMLDivElement | null>(null);
+
+  const { rows } = table.getRowModel();
+  const rowVirtualizer = useVirtualizer({
+    count: rows.length,
+    getScrollElement: () => tableContainerRef.current,
```

Comment:
For the ref, we may need to look into passing a ref as props if the user of PaginatedTable wants to use a different container ref. In our case, we'd use the whole page as they scroll.